plugins/language_ring.lua

```diff
@@ -0,0 +1,123 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Ring",
+  files = { "%.ring$", "%.rh$", "%.rform$" },
+  comment = "//",
+  patterns = {
+    { pattern = "#.-\n",                  type = "comment"  },
+    { pattern = "//.-\n",                 type = "comment"  },
+    { pattern = { "/%*", "%*/" },         type = "comment"  },
+    { pattern = { '"', '"', '\\' },       type = "string"   },
+    { pattern = { "'", "'", '\\' },       type = "string"   },
+    { pattern = { "`", "`" },             type = "string"   },
+    { pattern = "-?%d+[%d%.]*f?",         type = "number"   },
+    { pattern = "-?0x%x+",                type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",   type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",       type = "function" },
+    { pattern = "[%a_][%w_]*",            type = "symbol"   },
+    { pattern = ":%a+",                   type = "literal" },
+  },
+  symbols = {
```

Comment:
It looks like the language is not case-sensitive. Would it be a good idea to add some casing options for these?
So for example `see`, `See` and `SEE`.
You could loop over the various symbols and add the various other cases programmatically.

Reply:
Hi @Guldoman, added it in this commit [9c0e059](https://github.com/lite-xl/lite-xl-plugins/pull/487/commits/9c0e059a0dd91e46546a11cf9b66d1c7f4012772)